Library/Homebrew/download_strategy.rb

```diff
@@ -473,17 +473,8 @@ def stage
 # distribution.  (It will work for public buckets as well.)
 class S3DownloadStrategy < CurlDownloadStrategy
   def _fetch
-    # Put the aws gem requirement here (vs top of file) so it's only
-    # a dependency of S3 users, not all Homebrew users
-    require "rubygems"
-    begin
-      require "aws-sdk-v1"
-    rescue LoadError
-      onoe "Install the aws-sdk gem into the gem repo used by brew."
-      raise
-    end
-
-    if @url !~ %r{^https?://([^.].*)\.s3\.amazonaws\.com/(.+)$}
+    unless @url =~ %r{^https?://([^.].*)\.s3\.amazonaws\.com/(.+)$} ||
+           @url =~ %r{^s3://([^.].*?)/(.+)$}
```

Comment:
Could you keep the changes like this into a separate PR? It makes it easier to review and test this one. Thanks!

Reply:
Sure